pkgs/top-level/all-packages.nix

```diff
@@ -4823,27 +4823,29 @@ in
 
   phodav = callPackage ../tools/networking/phodav { };
 
-  pinentry = callPackage ../tools/security/pinentry {
+  pinentry-real = callPackage ../tools/security/pinentry {
```

Comment:
Any reason for doing this? I'd prefer to not introduce any new names like this.

Reply:
See https://github.com/NixOS/nixpkgs/pull/55061#issuecomment-460108869 and above discussion.

Reply:
I guess I'd rather have res. than have to deal with the -real derivations. You could also fix the no-x-libs.nix thing by just overriding like:

```
pinentry = super.pinentry.override {
  gtk2 = null;
}
```

in no-x-libs.nix

Reply:
i suppose this isn't so bad for now. Could you call it pinentry_full to a be a little more descriptive?

Reply:
What worries me though is having to have a "real" version for every package with an override. That looks like it is needed for the ```a = a_<flavor>;``` type overrides to work.

Reply:
Another solution would be to change this line in no-x-libs.nix from

```
      pinentry = super.pinentry_ncurses;
```

to

```
pinentry = super.pinentry.override { gtk2 = null; }
```

Reply:
This has been replaced by a multiple outputs on a single derivation. While that's horrible for _derivation_ closure size and rebuilds, it makes this discussion irrelevant for now.